File: Core/Entities/BirthdayOrders/BirthdayPackage.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Core.Entities.Discounts;

namespace Core.Entities.BirthdayOrders
{
    public class BirthdayPackage : BaseEntity
    {
        [Required]
		[MaxLength(255)]
        public string PackageName { get; set; }

        [Required]
		[MaxLength(2000)]
        public string Description { get; set; }
        public int NumberOfParticipants { get; set; }
        public decimal Price { get; set; }
        public decimal AdditionalBillingPerParticipant { get; set; }
        public int Duration { get; set; }
        public string Picture { get; set; }
        public decimal? DiscountedPrice { get; set; }
        public bool? HasDiscountsApplied { get; set; }

        public ICollection<BirthdayPackageKidActivity> BirthdayPackageKidActivities { get; set; }
        public ICollection<BirthdayPackageDiscount> BirthdayPackageDiscounts { get; set; }
        public ICollection<ClientBirthdayOrder> ClientBirthdayOrders { get; set; }
    }
}

File: angular-happykids/src/app/shared/models/birthdaypackage.ts
import { Discount } from './discount';
import { KidActivity } from './kidactivity';

export interface BirthdayPackage {
    id: number;
    packageName: string;
    description: string;
    numberOfParticipants: number;
    price: number;
    additionalBillingPerParticipant: number;
    discountedAdditionalBillingPerParticipant: number;
    duration: number;
    picture: string;
    discountSum: number;
    discountedPrice?: number;
    hasDiscountsApplied?: boolean;
    kidActivities: KidActivity[];
}

export interface BirthdayPackageCreateEdit {
    id: number;
    packageName: string;
    description: string;
    numberOfParticipants: number;
    price: number;
    additionalBillingPerParticipant: number;
    duration: number;
    picture: File;
    discountsIds: number[];
    kidActivitiesIds: number[];
}

export interface BirthdayPackagePutGet {
    BirthdayPackage: BirthdayPackage;
    selectedKidActivities: KidActivity[];
    nonSelectedKidActivities: KidActivity[];
    selectedDiscounts: Discount[];
    nonSelectedDiscounts: Discount[];
}
